pytensor/link/pytorch/dispatch/subtensor.py

```diff
@@ -0,0 +1,78 @@
+from pytensor.link.pytorch.dispatch.basic import pytorch_funcify
+from pytensor.tensor.subtensor import (
+    AdvancedIncSubtensor1,
+    AdvancedSubtensor,
+    AdvancedSubtensor1,
+    IncSubtensor,
+    Subtensor,
+    indices_from_subtensor,
+)
+from pytensor.tensor.type_other import MakeSlice
+
+
+@pytorch_funcify.register(Subtensor)
+def pytorch_funcify_Subtensor(op, node, **kwargs):
+    idx_list = getattr(op, "idx_list", None)
+
+    def subtensor(x, *ilists):
+        indices = indices_from_subtensor(ilists, idx_list)
+        for i in indices:
+            if isinstance(i, slice):
+                if i.step and i.step < 0:
+                    raise NotImplementedError(
+                        "Negative step sizes are not supported in Pytorch"
+                    )
```

Comment:
Make this checker a helper since it's needed for all Subtensor (basic, advanced, inc) Ops